tests/integration/test_services_intelligence_all.py

```
@@ -0,0 +1,1210 @@
+        assert removed == 1
+        assert engine.get_usage_data("stale_pat") is None
+        assert engine.get_usage_data("fresh_pat") is not None
+
+
+# ---------------------------------------------------------------------------
+# PatternScorer / ScoredPattern
+# ---------------------------------------------------------------------------
+
+
+class TestPatternScorer:
+    def _make_pattern(
```

Ruff (0.15.9)

[warning] Missing return type annotation for private function `_make_pattern`

(ANN202)